addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -147,6 +147,13 @@ def _check_date_from_date_to(self):
         if any(allocation.date_to and allocation.date_from > allocation.date_to for allocation in self):
             raise UserError(_("The Start Date of the Validity Period must be anterior to the End Date."))
 
+    def _check_employee_accrual_plan_based_on_worked_time(self):
```

Comment:
Could you please add a test to ensure the UserError is correctly raised + check it's not broken during forward port ?